Allow uploaded protected area to be given a name

ikas opened the pull request:
## Allow uploaded protected area to be given a name

### Overview

This PR adds the possibility of providing (optionally) a `name` when uploading shapefile for a project scenario using the endpoint `POST api/v1/scenarios/':id/protected-areas/shapefile')`.

If no `name` is provided, the current behavior is kept - the file name is used and inserted in the DB as `fullName`.

### Testing instructions

1. Call the `POST api/v1/scenarios/':id/protected-areas/shapefile')` endpoint, providing only the shapefile - the inserted row in the `wdpa` table in the `geo` database should have `fullName` set to the name of the uploaded file (existing behavior).
2. Call the `POST api/v1/scenarios/':id/protected-areas/shapefile')` endpoint, providing the shapefile and a `name` in the request body - the inserted row in the `wdpa` table in the `geo` database should have `fullName` set to the provided `name`.

### Feature relevant tickets

[MARXAN-923](https://vizzuality.atlassian.net/browse/MARXAN-923)

---

## Checklist before submitting

- [x] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

kgajowy commented on api/apps/geoprocessing/src/modules/protected-areas/dto/upload.shapefile.dto.ts:
```diff
@@ -0,0 +1,9 @@
+import { IsOptional, IsString } from 'class-validator';
+
+export class UploadShapefileDto {
+  @ApiPropertyOptional()
+  @IsString()
+  @IsOptional()
+  name?: string;
```
This can be potentially dangerous:

https://nodejs.org/en/knowledge/file-system/security/introduction/

implementation in our codebase:
`api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts`

kgajowy replied:
sql injection ?

kgajowy replied:
may not apply if using parametrized queries

kgajowy reviewed (changes requested):
Reasons: potential security issue; using -geo file within -api application (won't compile/run).

ikas commented on api/apps/geoprocessing/src/main.ts:
```diff
@@ -14,6 +14,6 @@ async function bootstrap() {
   );
   app.useGlobalFilters(new AllExceptionsFilter());
-  await app.listen(3000);
+  await app.listen(AppConfig.get('api.port', 3000));
```
Replaced api reference with geo and moved the commit into a separate PR, here: https://github.com/Vizzuality/marxan-cloud/pull/625

ikas commented on api/apps/api/src/modules/scenarios/scenarios.controller.ts:
```diff
@@ -91,6 +91,7 @@ import { asyncJobTag } from '@marxan-api/dto/async-job-tag';
 import { inlineJobTag } from '@marxan-api/dto/inline-job-tag';
 import { submissionFailed } from '@marxan-api/modules/scenarios/protected-area';
 import { ProtectedAreaDto } from '@marxan-api/modules/scenarios/dto/protected-area.dto';
+import { UploadShapefileDto } from '@marxan-geoprocessing/modules/protected-areas/dto/upload.shapefile.dto';
```
Hum, I think I just misplaced the DTO, it can live inside `api` as its only used there - will make that change :+1:

ikas commented on api/apps/api/src/modules/scenarios/scenarios.controller.ts:
```diff
@@ -238,7 +239,6 @@ export class ScenariosController {
         case projectDoesntExist:
           throw new NotFoundException(`Project doesn't exist`);
         default:
-          const _check: never = result.left;
```
Added back these lines - however, we should find a way to remove these IDE hints, or something, as this is misleading and someone (like me) might remove these unused variables unintendedly.

ikas commented:
@kgajowy can you give it another review please?

I've addressed all the feedback provided, and added a test (https://github.com/Vizzuality/marxan-cloud/pull/618/commits/c18abdf44c6bc7d3b08a37150750a30053231f22), let me know what you think :)
